### ISSUE TITLE
Update guide to cover entire process

### ISSUE THREAD

**Issue by gibfahn** [Trott/nodetodo#4]
I was talking to some of the other people who helped out at Code and Learn, and I think it would be good for the guidelines here to basically exactly specify the sequence of commands a contributor would need to run from `clone` to `Submit Pull Request`.

Linking to `BUILDING.md`, `CONTRIBUTING.md` and other docs is good, but I think repeating the same info is fine, there's a lot of other stuff in there that makes it hard to parse for a first contribution.

### First stab at a flow:

A lot of this is already on the [getting started](http://nodetodo.org/getting-started/) page.

- Create Github Account
- Fork Node
- Clone node
- `cd node`
- `git remote add upstream`
- git config set user.name/email
- Install dependencies (make/gcc or vcbuildtools/GitBash)
- `make` or `vcbuild`
- `make test` or `vcbuild nobuild test`
- Stop and ask for an issue (if it passes), ask for help (if it fails). Don't feel you need to debug yourself.
- `git checkout -b my-feature-name`
- Make changes required
- `make test`/`vcbuild nobuild test` (you can use `make lint`/`vcbuild nobuild lint` for just linting, or `tools/test.py test/parallel/test-name.js` for just one test)
- Get everything green
- `git status` (run after every command so you know what's going on)
- `git add /path/to/files/changed`
- `git commit`
- Follow commit message guidelines [here], example commit message: <> (list key things to remember)
- `npm i -g core-validate-commit; core-validate-commit --no-validate-metadata HEAD` (make sure your commit message is okay, run `git commit --amend` to edit the commit message).
- `git push -u origin my-feature-name`
- Go to `nodejs/node` and refresh, should see `Create Pull Request` button (o/w go to ...)
- Tick tickboxes by changing `- [ ]` to `- [x]`, write `Code and learn` in comment. Remove irrelevant tickbox lines.
- Click `Submit Pull Request`
- Raise your hand in code-and-learn or cc/ @nodejs/mentors (doesn't exist yet) otherwise
- Check back, and if someone requests a change then:
  - Make changes
  - Make sure tests pass
  - `git commit -a`
  - `git push`
  - Reply saying "Done, PTAL"

**Comment by mscdex** [comment]
I think [`core-validate-commit`](https://github.com/nodejs/core-validate-commit) should be added here and used prior to PR submissions. The reason is that most of these changes being made are straight forward and will not involve temporary squash commits and the like, and it will possibly save some time for those merging the PRs.

**Comment by gibfahn** [comment]
>I think core-validate-commit should be added here and used prior to PR submissions.

Problem is that it fails, so we have to tell people that "passing" looks like this:

<img width="540" alt="image" src="https://user-images.githubusercontent.com/15943089/32452495-9674a802-c311-11e7-9d5b-e9a841c86282.png">

Really we need an option to `core-validate-commit` that ignores those two tests.

**Comment by Trott** [comment]
> Really we need an option to core-validate-commit that ignores those two tests.

@gibfahn `core-validate-commit  --no-validate-metadata`

**Comment by ryanmurakami** [comment]
I don't even know why I'm typing this.

**Comment by Trott** [comment]
I'm happy to accept pull requests to improve the getting-started page, but for now, I'm going to close this. If anyone thinks they want to do some of this stuff and plan to submit one or more pull requests, please comment. Thanks!
